py/kubeflow/kubeflow/ci/workflow_utils.py

```diff
@@ -200,14 +202,17 @@ def create_kaniko_task(self, task_template, dockerfile, context,
         extend the code.
         """
         kaniko = argo_build_util.deep_copy(task_template)
+        # for shurt UUID generation
+        alphabet = string.ascii_lowercase + string.digits
 
         # append the tag base-commit[0:7]
         if ":" not in destination:
             sha = os.getenv("PULL_BASE_SHA", "12341234kanikotest")
             base = os.getenv("PULL_BASE_REF", "master")
             destination += ":%s-%s" % (base, sha[0:8])
 
-        kaniko["name"] = "kaniko-build-push"
+        # add short UUID to step name to ensure it is unique
+        kaniko["name"] = "kaniko-build-push-" + ''.join(random.choices(alphabet, k=8))
```

Comment:
Could you provide some more details as to why this change is needed?

Reply:
Otherwise there are 2 steps in the workflow (this workflow builds 2 images) that are both named `kaniko-build-push`. I wasn't sure if it would like that so I tested it and Argo does not :).

Reply:
The same thing is done here BTW: https://github.com/kubeflow/kubeflow/pull/5626/files#diff-a58831ea15b3ba92fe93b2196dc2867189dc87c3836c617833dbdf652ad22fd0R205

Reply:
Here you can see what the workflow I just linked looks like when it runs. 
https://argo.kubeflow-testing.com/workflows/kubeflow-test-infra/kubeflow-kubeflow-presubmit-nb-j-pt-5626-5feb726-6224-63fc?tab=workflow&nodeId=kubeflow-kubeflow-presubmit-nb-j-pt-5626-5feb726-6224-63fc-2134638194

Reply:
Makes sense